verilog/tools/ls/verilog-language-server_test.cc

```diff
@@ -0,0 +1,510 @@
+  ASSERT_EQ(diagnostics_parsed["params"]["diagnostics"].size(), 0)
+      << "The test file has errors";
+  const absl::string_view highlight_request1 =
+      R"({"jsonrpc":"2.0", "id":20, "method":"textDocument/documentHighlight","params":{"textDocument":{"uri":"file://sym.sv"},"position":{"line":1,"character":7}}})";
+  Request(highlight_request1);
+  status = ServerStep();
+  ASSERT_TRUE(status.ok()) << status;
+
+  std::string highlight_response1;
+  Response(&highlight_response1);
+
+  json highlight_response1_parsed = json::parse(highlight_response1);
+  ASSERT_EQ(highlight_response1_parsed["id"], 20);
+  ASSERT_EQ(highlight_response1_parsed["result"].size(), 2);
+  ASSERT_EQ(
+      highlight_response1_parsed["result"][0],
+      json::parse(
+          R"({"range":{"start":{"line":1, "character": 7}, "end":{"line":1, "character": 8}}})"));
+  ASSERT_EQ(
+      highlight_response1_parsed["result"][1],
+      json::parse(
+          R"({"range":{"start":{"line":1, "character": 20}, "end":{"line":1, "character": 21}}})"));
+
+  const absl::string_view highlight_request2 =
+      R"({"jsonrpc":"2.0", "id":21, "method":"textDocument/documentHighlight","params":{"textDocument":{"uri":"file://sym.sv"},"position":{"line":1,"character":2}}})";
+  Request(highlight_request2);
+  status = ServerStep();
+  ASSERT_TRUE(status.ok()) << status;
+
+  std::string highlight_response2;
+  Response(&highlight_response2);
+
+  json highlight_response2_parsed = json::parse(highlight_response2);
+  ASSERT_EQ(highlight_response2_parsed["id"], 21);
+  ASSERT_EQ(highlight_response2_parsed["result"].size(), 0);
+}
+
+// Tests structure holding data for test textDocument/rangeFormatting requests
+struct FormattingRequestParams {
+  FormattingRequestParams(int id, int start_line, int start_character,
+                          int end_line, int end_character,
+                          absl::string_view new_text, int new_text_start_line,
+                          int new_text_start_character, int new_text_end_line,
+                          int new_text_end_character)
+      : id(id),
+        start_line(start_line),
+        start_character(start_character),
+        end_line(end_line),
+        end_character(end_character),
+        new_text(new_text),
+        new_text_start_line(new_text_start_line),
+        new_text_start_character(new_text_start_character),
+        new_text_end_line(new_text_end_line),
+        new_text_end_character(new_text_end_character) {}
+
+  int id;
+  int start_line;
+  int start_character;
+  int end_line;
+  int end_character;
+
+  absl::string_view new_text;
+  int new_text_start_line;
+  int new_text_start_character;
+  int new_text_end_line;
+  int new_text_end_character;
+};
+
+// Creates a textDocument/rangeFormatting request from FormattingRequestParams
+// structure
+std::string FormattingRequest(absl::string_view file,
+                              const FormattingRequestParams &params) {
+  json formattingrequest = {
```

Comment:
Maybe it would be easier to use the structs that are generated in `lsp-protocol.h` ?
(We can tackle that in a future pull request, this is a good starting point.)